API-strategie-extensies/Summary v1.0.md

```diff
@@ -44,6 +44,18 @@ Design rules have unique and permanent numbers. In the event of design rules bei
 * <a href="#api-50">API-50</a>: Use CORS to control access
 * <a href="#api-52">API-52</a>: Use OAuth 2.0 for authorization with rights delegation
 * <a href="#api-58">API-58</a>: Do not put any sensitive information in URIs when communicating over shared networks
+* <a href="#api-59">API-59</a>: Use spinal-case for path segments
+* <a href="#api-60">API-60</a>: Normalize characters with diacritics to their base characters for path segments
+* <a href="#api-61">API-61</a>: Do not explicitly indicate that a resource is an API
+* <a href="#api-62">API-62</a>: Do not use file extensions in path segments
+* <a href="#api-63">API-63</a>: Do not use nonstandard abbreviations as resource names
+* <a href="#api-64">API-64</a>: Do not use compound words for nested objects
+* <a href="#api-65">API-65</a>: Use enumerations only for fixed sets of values that will not change
```

Comment:
Will not change is a high bar to clear... if you want to include this it should be something along the lines of 'changing enumerations should involve versioning of the API'...  

Or maybe "Use enumerations only for fixed sets of values that will not change within an API version"

Reply:
Thanks @mtrimpe . If the values are likely to change at some stage, an enum should not be used. But I'll try to find a different way to phrase this message differently..

Reply:
naar mijn idee is het een afweging tussen de noodzaak om het als enumeratie te gebruiken en het risico dat de lijst mogelijke waarden wijzigt. 
Bij noodzaak om als enumeratie op te nemen denk ik aan:
- zorgen dat de client weet welke waarde gebruikt kan worden (in PUT of POST body of query parameter)
- wanneer de client de waarde moet gebruiken in een algoritme (if (status=="VERBLIJFSOBJECT_IN_GEBRUIK") then ...)

Dus het enkele feit dat je verwacht dat de lijst erg stabiel is, is niet voldoende reden om het als enumeratie te definiëren.
En andersom wanneer de client voor gebruik van de API de mogelijke waarden moet weten en in code gebruiken, moet je wellicht toch een enumeratie gebruiken. Alleen mag koppeling tussen enumeratie in de bron en enumeratie in de API niet direct zijn, dit moet losgekoppeld worden. Dan moet je er dus achter komen welke enumeratiewaarden voor de gebruikers relevant zijn, grote kans dat die lijst veel minder vaak wijzigt.